Formula/ccache.rb

```diff
@@ -28,8 +28,14 @@ class Ccache < Formula
   fails_with gcc: "5"
 
   def install
-    system "cmake", ".", *std_cmake_args
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_IPO=TRUE"
+    system "cmake", "--build", "build"
+
+    # Homebrew compiler shim actively prevents ccache usage (see caveats), which will break the test suite
+    system "cmake", "-E", "env", "CC=/usr/bin/#{DevelopmentTools.default_compiler}",
+           "ctest", "-j#{ENV.make_jobs}", "--test-dir", "build"
```

Comment:
```suggestion
    with_env(CC: DevelopmentTools.locate(ENV.cc)) do
      system "ctest", "-j#{ENV.make_jobs}", "--test-dir", "build"
    end        
```
I think this should work too and is more consistent with how we override environment variables.

Not sure if using system GCC (`gcc-5`) is the intention on Linux, but this also changes the test to use Homebrew GCC. If we want to keep testing with system GCC, just pass `DevelopmentTools.default_compiler` to `DevelopmentTools.locate`. (Though that is a lot more verbose now...)

Reply:
Or maybe not. I don't remember if we have an audit that rules out calling `ctest`.

Reply:
Ah, I knew there was something like `with_env`! I just didn't know what to search for to find it. Agree that this is more idiomatic Homebrew-wise.